File: src/main/java/com/everyonewaiter/domain/sse/SseKeyType.java
package com.everyonewaiter.domain.sse;

public enum SseKeyType {

  EMITTER,
  EVENT,
  ;

  private static final String KEY_DELIMITER = ":";

  public String createKey(String prefix) {
    return createScanKey(prefix) + System.currentTimeMillis();
  }

  public String createScanKey(String prefix) {
    return prefix + KEY_DELIMITER + getLowerCaseName() + KEY_DELIMITER;
  }

  private String getLowerCaseName() {
    return name().toLowerCase();
  }

}

File: src/main/kotlin/com/everyonewaiter/common/sse/SseKeyType.kt
package com.everyonewaiter.common.sse

/**
 * SSE 이미터 및 이벤트 저장소에서 사용할 키를 구분하기 위한 열거형입니다.
 * 이미터 및 이벤트 저장소에서 사용되는 키의 포맷이 동일하기 때문에 구분이 필요합니다.
 */
enum class SseKeyType {
    EMITTER,
    EVENT,
    ;

    val lowerCaseName: String
        get() = name.lowercase()
}
